### ISSUE TITLE
Trim mixin block / block as a variable

### ISSUE THREAD

**Issue by tenry92** [jadejs/jade#2073]
I've got a problem with the current jade version, using with io.js.

I was trying to design a mixin, which I could use in this way:
```jade
+form-control('title', 'text')= title
+form-control('description', 'textarea')= description
```

And my implementation looks like this:
```jade
mixin form-control(name, type)
  case type
    default
      input.form-control(type=type, name=name, value=block)&attributes(attributes)
    when 'textarea'
      textarea.form-control(name=name)
        block
```

First, I get a strange value for my normal text input, something with 'function (){buf.push...'.
The second one, textarea, works better, but I get a line break + indentation before the <textarea> closes, and this gives extra characters to the actual value.

**Comment by TimothyGu** [comment]
tl;dr: the syntax (block as a variable) is not supported.

For now, you can use `value=block()`. But this usage is an implementation detail, is not officially supported and may be broken in the next version of Jade.
